### PR TITLE
Checkmarx

### PR DESCRIPTION
**Pull request by geoffwhittington**

### CODE REVIEW

**Comment by geoffwhittington** [comment]
@ehsanf @funkaoshi Here's the Checkmarx file upload stuff. I'll have a separate MR for SDE for the frontend stuff

File: sdetools/modules/import_checkmarx/tools/map_transform.py [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+                tasks = []
+                if cwe_mapping.has_key(cwe.attributes['id'].value):
+                    tasks = cwe_mapping[cwe.attributes['id'].value]
+                tasks.append(task.attributes['id'].value)
+                cwe_mapping[cwe.attributes['id'].value] = tasks
+
+        self.mapping = cwe_mapping
+
+    def remap(self, unmapped_cwe_file_name):
+        fp = open(unmapped_cwe_file_name, "w")
```
Comment (funkaoshi) [inline sdetools/modules/import_checkmarx/tools/map_transform.py]: Should probably also use `with open(...)` here as well.

File: sdetools/modules/import_checkmarx/tools/map_transform.py [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+                weaknesses = []
+                if cwe_checks.has_key(check['cwe_id']):
+                    weaknesses = cwe_checks[check['cwe_id']]
+                weaknesses.append(check)
+                cwe_checks[check['cwe_id']] = weaknesses
+
+            self.cwe_checks = cwe_checks
+
+    def output_mapping(self, file_name):
+        fp = open(file_name, "w")
```
Comment (funkaoshi) [inline sdetools/modules/import_checkmarx/tools/map_transform.py]: And here as well. (`with open(...)`)

Comment (geoffwhittington) [inline sdetools/modules/import_checkmarx/tools/map_transform.py]: This file is to help us build the mapping file based on other inputs - used maybe a couple of times a year maximum, and may even get deleted when this mapping is completed. Its here just for tracking purposes

File: sdetools/modules/import_checkmarx/tools/map_transform.py [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+                return True
+        return False
+
+    def check_mapped(self, task_map, check_id, selected_task=None):
+        keys = task_map.iterkeys()
+
+        if self.check_in_list(self.thirdparty_checks, check_id):
+            return True
+
+        for task_id in keys:
```
Comment (funkaoshi) [inline sdetools/modules/import_checkmarx/tools/map_transform.py]: I'd define `keys`, here, or just loop over `task_map.iterkeys()` directly. You aren't using `keys` anywhere else.

**Pull request merged**